drivers/netdev_ieee802154/netdev_ieee802154.c

```diff
@@ -50,6 +50,19 @@ static int _get_iid(netdev_ieee802154_t *dev, eui64_t *value, size_t max_len)
     return sizeof(eui64_t);
 }
 
+void netdev_ieee802154_reset(netdev_ieee802154_t *dev)
+{
+    dev->seq = 0;
```

Comment:
Maybe use a random number here instead?

Reply:
I don't disagree with you, but that would be a change in behaviour for the radio drivers here. Would you mind if I do that as a small follow up to keep this PR purely as cleanup?

Reply:
:+1:

Reply:
I haven't found any usage of `seq` in one of the touched drivers, its used in the higher gnrc layers, but they are driver independent already. Hence, there should be no effect on the driver behaviour right now - or did I miss something?

Reply:
@smlng No, you're right, this should not affect the drivers. 

In the future I want to move the full `seq` handling to `netdev_ieee802154`.

Reply:
It is specific for ieee802154 (matching ACKs to frames), so I'm afraid moving it to `netdev` doesn't make sense here.

Reply:
as said: just a question - I'm fine with keeping it .15.4 specific.

Reply:
btw. what do you mean by moving the `full handling of seq` to netdev_ieee802154? You mean the gnrc stuff?

Reply:
The problem I'm facing at the moment with implementing #7736 is that gnrc_netif_ieee802154 directly accesses data from the netdev struct. The problem with this is that gnrc_netif_ieee802154 assumes what kind of netdev device/layer is used. As long as this direct access is there, it is not possible to use anything else than a netdev_iee802154_t layer for the netif->dev. One possible solution for this is to move the handling of the `seq` number to the netdev_ieee802154 code.